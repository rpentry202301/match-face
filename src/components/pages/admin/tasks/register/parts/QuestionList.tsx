'use client'
import CheckBox from "@/components/ui/checkbox/CheckBox"
import { ChangeEvent } from "react"

type Question = {
  id: number,
  projectId: number,
  projectName: string,
  context: string,
  updateAt: string,
}

const QuestionList = ({
  questions,
  checkedValues,
  onChange
}: {
  questions: Question[],
  checkedValues: string[],
  onChange: (e: ChangeEvent<HTMLInputElement>) => void
}) => {
  const tableDefaultClassName = "border-2 border-deep-gray px-4 py-2 text-center"
  return (
    <div className="w-full mx-auto">
      <table className="border-collapse border-2 border-deep-gray text-sm mx-auto w-full">
        <thead className="bg-light-gray">
          <tr>
            <th className={`${tableDefaultClassName} w-1/12`}></th>
            <th className={`${tableDefaultClassName} w-2/12`}>最終編集日</th>
            <th className={`${tableDefaultClassName} w-3/12`}>案件名</th>
            <th className={`${tableDefaultClassName}`}>質問内容</th>
          </tr>
        </thead>
          <tbody>
            {questions.map((question) => (
              <tr key={`qId_${question.id}`}>
                <td className={`${tableDefaultClassName}`}>
                  <CheckBox
                    value={question.context}
                    onChange={onChange}
                    checked={checkedValues.includes(question.context)}
                  />
                </td>
                <td className={`${tableDefaultClassName}`}>{question.updateAt.slice(0, 10)}</td>
                <td className={`${tableDefaultClassName}`}>{question.projectName}</td>
                <td className={`${tableDefaultClassName}`}>{question.context}</td>
              </tr>
            ))}
          </tbody>
      </table>
    </div>
  )
}

export default QuestionList
