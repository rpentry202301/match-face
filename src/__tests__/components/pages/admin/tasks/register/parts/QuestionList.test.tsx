import QuestionList from "@/components/pages/admin/tasks/register/parts/QuestionList";
import { render } from "@testing-library/react";

const questionList = [
  {
    id: 1,
    projectId: 1,
    projectName: 'バックエンド案件',
    context: '開発経験について教えてください。',
    createdUser: '',
    createdAt: '-999999999-01-01T00:00:00',
    updateUser: 'テスト花子',
    updateAt: '2023-09-07T09:29:24.412579'
  },
  {
    id: 2,
    projectId: 1,
    projectName: 'バックエンド案件',
    context: '1番得意な言語は何ですか。',
    createdUser: '',
    createdAt: '-999999999-01-01T00:00:00',
    updateUser: 'テスト花子',
    updateAt: '2023-09-07T09:29:24.412579'
  }
]

describe("Question.tsx", () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });
  afterEach(() => {
    jest.clearAllMocks();
  });

  it("QuestionListのスナップショット", () => {
    const { container } = render(
      <QuestionList
        questions={questionList}
        checkedValues={["開発経験について教えてください。"]}
        onChange={jest.fn()}
      />
    
    );
    expect(container).toMatchSnapshot();
  });
})
