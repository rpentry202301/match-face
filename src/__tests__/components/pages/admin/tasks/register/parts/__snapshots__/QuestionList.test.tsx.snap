// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Question.tsx QuestionListのスナップショット 1`] = `
<div>
  <div
    class="w-full mx-auto"
  >
    <table
      class="border-collapse border-2 border-deep-gray text-sm mx-auto w-full"
    >
      <thead
        class="bg-light-gray"
      >
        <tr>
          <th
            class="border-2 border-deep-gray px-4 py-2 text-center w-1/12"
          />
          <th
            class="border-2 border-deep-gray px-4 py-2 text-center w-2/12"
          >
            最終編集日
          </th>
          <th
            class="border-2 border-deep-gray px-4 py-2 text-center w-3/12"
          >
            案件名
          </th>
          <th
            class="border-2 border-deep-gray px-4 py-2 text-center"
          >
            質問内容
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td
            class="border-2 border-deep-gray px-4 py-2 text-center"
          >
            <input
              checked=""
              class="w-4 h-4 accent-orange cursor-pointer"
              type="checkbox"
              value="開発経験について教えてください。"
            />
          </td>
          <td
            class="border-2 border-deep-gray px-4 py-2 text-center"
          >
            2023-09-07
          </td>
          <td
            class="border-2 border-deep-gray px-4 py-2 text-center"
          >
            バックエンド案件
          </td>
          <td
            class="border-2 border-deep-gray px-4 py-2 text-center"
          >
            開発経験について教えてください。
          </td>
        </tr>
        <tr>
          <td
            class="border-2 border-deep-gray px-4 py-2 text-center"
          >
            <input
              class="w-4 h-4 accent-orange cursor-pointer"
              type="checkbox"
              value="1番得意な言語は何ですか。"
            />
          </td>
          <td
            class="border-2 border-deep-gray px-4 py-2 text-center"
          >
            2023-09-07
          </td>
          <td
            class="border-2 border-deep-gray px-4 py-2 text-center"
          >
            バックエンド案件
          </td>
          <td
            class="border-2 border-deep-gray px-4 py-2 text-center"
          >
            1番得意な言語は何ですか。
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
